### ISSUE TITLE
introduce hotfix capability 

### ISSUE THREAD

**Issue by gnadelwartz** [qooob/authentic-theme#1069]
As we had a bug in cron (and now maybe in spam) which leads to a 1.881 release
and I requested to update `update-from-repo.sh` with authentic updates also, it would be nice to have a general hotfix possiblility.

e.g. if we have a typo in a script which can be fixed by simply downloading the correct version of the script, this may be done as a hotfix with 'update-from-repo.sh -file spam/save_report.cgi' or similar ...

**Comment by gnadelwartz** [comment]
typical example would be this fix https://github.com/webmin/webmin/pull/875 where I forget a $ for a variable name.

**Comment by qooob** [comment]
Okay, so you want it to run where in theme? Make a patch applied based on module name?

**Comment by qooob** [comment]
Like open UI, it would ask you which module do you want to update. It would list that modules and you just check them and hit okay button?

**Comment by gnadelwartz** [comment]
correct known errors which may affect many users and can't wait until next release.
So more a selected/curated list of hotfixes, not simply a pull module from dev

**Comment by gnadelwartz** [comment]
lets brainstorm:

- is this needed?
- how should it be implemented in UI
- who provides the hotfixes? (IMHO Jamie should judge)
- ...

**Comment by qooob** [comment]
Do we have a list of hot fixes to look at?

**Comment by gnadelwartz** [comment]
not now, beacuse we did not thick about this, but a canditae is: https://github.com/webmin/webmin/pull/875

**Comment by gnadelwartz** [comment]
on unix we have the command pacth which is designed for incremental updates: 'patch'
https://www.thegeekstuff.com/2014/12/patch-command-examples

**Comment by gnadelwartz** [comment]
First question to answer: Is this really needed?
I'm think it may be helpful. for real critical fixes Jamie will provide a new release, but this is more work than a small fix.

**Comment by gnadelwartz** [comment]
If you think its useless, I'll drop the idea.
If not, I think about a more elaborated proposal and open an issue in webmin ..

**Comment by gnadelwartz** [comment]
Why not use existing module installation/upgrade function for this? `*.wbm` files are simply renamed `*.tar.gz` files

I'll investigate if `*.wbm` files can be used for patch updates ...

**Comment by qooob** [comment]
I meant to check only those that needs an update, and update theme.

**Comment by gnadelwartz** [comment]
yes, but these are from dev, may have new untestested or broken functions, I think to do this for stable hotfixes, not directly from dev.

**Comment by qooob** [comment]
You could title commit having `hotfix` keyword?

**Comment by gnadelwartz** [comment]
Done https://github.com/webmin/webmin/pull/879
